plugins/catalog-react/src/components/UserListPicker/UserListPicker.tsx

```diff
@@ -161,19 +161,46 @@ export const UserListPicker = ({
     [isOwnedEntity, isStarredEntity],
   );
 
+  const [selectedUserFilter, setSelectedUserFilter] = useState(
+    [queryParameters.user].flat()[0] ?? initialFilter,
+  );
+
   // To show proper counts for each section, apply all other frontend filters _except_ the user
   // filter that's controlled by this picker.
-  const [entitiesWithoutUserFilter, setEntitiesWithoutUserFilter] =
-    useState(backendEntities);
-  const totalOwnedUserEntities = entitiesWithoutUserFilter.filter(entity =>
-    ownedFilter.filterEntity(entity),
-  ).length;
-  const [selectedUserFilter, setSelectedUserFilter] = useState(
-    totalOwnedUserEntities > 0
-      ? [queryParameters.user].flat()[0] ?? initialFilter
-      : 'all',
+  const entitiesWithoutUserFilter = useMemo(
+    () =>
+      backendEntities.filter(
+        reduceEntityFilters(
+          compact(Object.values({ ...filters, user: undefined })),
+        ),
+      ),
+    [filters, backendEntities],
+  );
+
+  const filterCounts = useMemo<Record<string, number>>(
+    () => ({
+      all: entitiesWithoutUserFilter.length,
+      starred: entitiesWithoutUserFilter.filter(entity =>
+        starredFilter.filterEntity(entity),
+      ).length,
+      owned: entitiesWithoutUserFilter.filter(entity =>
+        ownedFilter.filterEntity(entity),
+      ).length,
```

Comment:
```suggestion
      starred: entitiesWithoutUserFilter.filter(starredFilter.filterEntity).length,
      owned: entitiesWithoutUserFilter.filter(ownedFilter.filterEntity).length,
```
Could save a few characters here 😁

Reply:
Hmm, I might hold off from making this change personally, since it means passing a [bunch of extra parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#parameters) to our `filterEntity` methods, and could introduce tricky bugs later if those methods are updated in the future to accept more.